bindata/oauth-apiserver/deploy.yaml

```diff
@@ -101,18 +101,36 @@ spec:
         - mountPath: /var/log/oauth-apiserver
           name: audit-dir
         livenessProbe:
-          initialDelaySeconds: 30
           httpGet:
             scheme: HTTPS
             port: 8443
             path: healthz
+          initialDelaySeconds: 0
+          periodSeconds: 10
+          timeoutSeconds: 1
+          successThreshold: 1
+          failureThreshold: 3
         readinessProbe:
-          failureThreshold: 10
           httpGet:
             scheme: HTTPS
             port: 8443
             path: readyz
```

Comment:
The PR title says "make apiserver readiness probe honor readyz", but it does honor readyz already, no? or did you mean to say "make apiserver startupProbe honor readyz"?

Reply:
> The PR title says "make apiserver readiness probe honor readyz", but it does honor readyz already, no? or did you mean to say "make apiserver startupProbe honor readyz"?

It did not honor it during shutdown because 10 failures times 10 seconds between attempts, means that ready would not be set to false before the graceful termination was already over.  So the pod disappeared, it didn't go ready=false.  This tightens it considerably.